odoo/http.py

```diff
@@ -1718,7 +1718,16 @@ def _save_session(self):
         # - It could allow session fixation attacks.
         cookie_sid = self.httprequest.cookies.get('session_id')
         if not sess.is_explicit and (sess.is_dirty or cookie_sid != sess.sid):
-            self.future_response.set_cookie('session_id', sess.sid, max_age=SESSION_LIFETIME, httponly=True)
+            if self.db and not self.env.cr._closed:
```

Comment:
```suggestion
            if self.env and not self.env.cr._closed:
```
there are some nasty corner cases where `request.db` is set but `request.env` is not